cmd/fetchd/cmd/stakereconciliationmigrate.go

```diff
@@ -0,0 +1,320 @@
+					})
+					continue
+				}
+
+				newFetchAddr, err := sdk.AccAddressFromBech32(registration.NativeAddress)
+				if err != nil {
+					errs = append(errs, fmt.Errorf("failed to parse bech32 address for new account %q: %w", registration.NativeAddress, err))
+					continue
+				}
+
+				var oldAcctBalance, newAcctBalance sdk.Coins
+				var oldBalanceIndex, newBalanceIndex int
+				for i, b := range bankGenesis.Balances {
+					if b.GetAddress().Equals(oldFetchAddr) {
+						oldAcctBalance = b.GetCoins()
+						oldBalanceIndex = i
+						if newAcctBalance == nil {
+							continue
+						}
+						break
+					}
+
+					if b.GetAddress().Equals(newFetchAddr) {
+						newAcctBalance = b.GetCoins()
+						newBalanceIndex = i
+						if oldAcctBalance == nil {
+							continue
+						}
+						break
+					}
+				}
+
+				if oldAcctBalance == nil {
+					errs = append(errs, fmt.Errorf("old account with address %q does not have a balance", oldFetchAddr.String()))
+					continue
+				}
+
+				var oldAccount, newAccount authtypes.GenesisAccount
+				for _, _account := range genesisAccounts {
+					if _account.GetAddress().Equals(oldFetchAddr) {
+						oldAccount = _account
+						if newAccount == nil {
+							continue
+						}
+						break
+					}
+
+					if _account.GetAddress().Equals(newFetchAddr) {
+						newAccount = _account
+						if oldAccount == nil {
+							continue
+						}
+						break
+					}
+
+				}
+
+				if oldAccount == nil {
+					ineligible = append(ineligible, IneligibleRegistration{
+						NativeAddr: oldFetchAddr.String(),
+						Reason:     fmt.Errorf("unable to find old account with address %q", oldFetchAddr.String()),
+					})
+					continue
+				}
+
+				if newAccount == nil {
+					ineligible = append(ineligible, IneligibleRegistration{
+						NativeAddr: oldFetchAddr.String(),
+						Reason:     fmt.Errorf("unable to find new account with address %q", newFetchAddr.String()),
+					})
+					continue
+				}
+
+				if newAcctBalance == nil {
+					errs = append(errs, fmt.Errorf("new account with address %q does not have a balance", newFetchAddr.String()))
+					continue
+				}
+
+				if oldAccount.GetSequence() != 0 {
+					ineligible = append(ineligible, IneligibleRegistration{
+						NativeAddr: oldFetchAddr.String(),
+						Reason:     errNonZeroSeqNum,
+					})
+					continue
+				}
+
+				migrateCoin, err := sdk.ParseCoinNormalized(fmt.Sprintf("%s%s", row[3], coinDenom))
+				if err != nil {
+					errs = append(errs, fmt.Errorf("unable to parse amount of tokens to migrate for row %q: %w", row, err))
+					continue
+				}
+				migrateCoins := sdk.NewCoins(migrateCoin)
+
+				if !oldAcctBalance.IsEqual(migrateCoins) {
+					ineligible = append(ineligible, IneligibleRegistration{
+						NativeAddr: oldFetchAddr.String(),
+						Reason:     errBalanceNotMatch,
+					})
+					continue
+				}
+
+				// Zero out old account balance
+				bankGenesis.Balances[oldBalanceIndex].Coins = sdk.NewCoins()
```

Comment:
Do we need to worry about wiping out tokens of other denominations?

Reply:
right, we can add an extra safety test that we have only 1 denom to make sure we don't wipe another by mistake, it's unlikely but good to have